pkg/artifactory/resource/configuration/resource_artifactory_property_set.go

```diff
@@ -0,0 +1,290 @@
+	Properties []Property `xml:"properties>property" yaml:"properties"`
+}
+
+type PropertySets struct {
+	PropertySets []PropertySet `xml:"propertySets>propertySet" yaml:"propertySet"`
+}
+
+func ResourceArtifactoryPropertySet() *schema.Resource {
+	var propertySetsSchema = map[string]*schema.Schema{
+		"name": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      `Property set name.`,
+		},
+		"visible": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: `Defines if the list visible and assignable to the repository or artifact.`,
+		},
+		"property": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			Description: `A list of properties that will be part of the property set.`,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:             schema.TypeString,
+						Required:         true,
+						Description:      `The name of the property.`,
+						ValidateDiagFunc: validator.StringIsNotEmpty,
+					},
+					"closed_predefined_values": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+						Description: `Disables "multiple_choice" if set to "false" at the same time with multiple_choice set to "true".`,
+					},
+					"multiple_choice": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+						Description: `Whether or not user can select multiple values. "closed_predefined_values" should be set to "true".`,
+					},
+					"predefined_value": {
+						Type:        schema.TypeSet,
+						Required:    true,
+						Description: `Properties in the property set.`,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"name": {
+									Type:             schema.TypeString,
+									Required:         true,
+									Description:      `Predefined property name.`,
+									ValidateDiagFunc: validator.StringIsNotEmpty,
+								},
+								"default_value": {
+									Type:        schema.TypeBool,
+									Required:    true,
+									Description: `Whether the value is selected by default in the UI.`,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	var resourcePropertySetRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		propertySetConfigs := &PropertySets{}
+		// Unpacking HCL to compare the names of the property sets with the XML data we will get from the API
+		unpackedPropertySet := unpackPropertySet(d)
+
+		_, err := m.(*resty.Client).R().SetResult(&propertySetConfigs).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		var matchedPropertySet = PropertySet{}
+
+		matchedPropertySet = PropertySet{}
```

Comment:
This is redundant. Line 110 already initialize `matchedPropertySet` with an empty struct.

Reply:
removed